pydantic/fields.py

```diff
@@ -36,16 +51,117 @@ class Validator(NamedTuple):
+    schema will have a ``maximum`` validation keyword
+    :param min_length: only applies to strings, requires the field to have a minimum length. The
+    schema will have a ``maximum`` validation keyword
+    :param max_length: only applies to strings, requires the field to have a maximum length. The
+    schema will have a ``maxLength`` validation keyword
+    :param regex: only applies to strings, requires the field match agains a regular expression
+    pattern string. The schema will have a ``pattern`` validation keyword
+    :param **extra: any additional keyword arguments will be added as is to the schema
     """
 
-    __slots__ = 'default', 'alias', 'title', 'extra'
+    __slots__ = (
+        'default',
+        'alias',
+        'title',
+        'description',
+        'gt',
+        'ge',
+        'lt',
+        'le',
+        'min_length',
+        'max_length',
+        'regex',
+        'extra',
+    )
 
-    def __init__(self, default, *, alias=None, title=None, **extra):
+    def __init__(
+        self,
+        default,
+        *,
+        alias: str = None,
+        title: str = None,
+        description: str = None,
+        gt: float = None,
+        ge: float = None,
+        lt: float = None,
+        le: float = None,
+        min_length: int = None,
+        max_length: int = None,
+        regex: str = None,
+        **extra,
+    ):
         self.default = default
         self.alias = alias
         self.title = title
+        self.description = description
         self.extra = extra
+        self.gt = gt
+        self.ge = ge
+        self.lt = lt
+        self.le = le
+        self.min_length = min_length
+        self.max_length = max_length
+        self.regex = regex
+
+
+_numeric_types = (int, float, Decimal)
+_blacklist = (EmailStr, DSN, UrlStr, ConstrainedStr, ConstrainedInt, ConstrainedFloat, ConstrainedDecimal)
+_str_attrs = ('max_length', 'min_length', 'regex')
+_numeric_attrs = ('gt', 'lt', 'ge', 'le')
+_map_types_const = {str: constr, int: conint, float: confloat, Decimal: condecimal}
+
+
+def get_annotation_from_schema(annotation, schema):
+    """Get an annotation with validation implemented for numbers and strings based on the schema.
+
+    :param annotation: an annotation from a field specification, as ``str``, ``ConstrainedStr`` or the
+      return value of ``constr(max_length=10)``
+    :param schema: an instance of Schema, possibly with declarations for validations and JSON Schema
+    :return: the same ``annotation`` if unmodified or a new annotation with validation in place
+    """
+    kwargs = {}
+    params_to_set = False
+    if (
+        isinstance(annotation, type)
+        and issubclass(annotation, (str,) + _numeric_types)
+        and not issubclass(annotation, bool)
+        and not any([issubclass(annotation, c) for c in _blacklist])
+    ):
+        if issubclass(annotation, str):
```

Comment:
Yeah, but as I'm calling `constr` to generate the final type, it would always output `str`, instead of `bytes`.

I guess one option would be adding a `conbytes`, or using the validators directly, like `anystr_length_validator`, but I wanted to use internally the same functions that already exist.

What do you think we should do about that?